crates/codegen/ebnf/src/ebnf_writer.rs

```diff
@@ -0,0 +1,76 @@
+#[derive(Debug)]
+#[allow(non_camel_case_types)]
+pub enum TokenKind {
+    comment,
+    constant,
+    keyword,
+    operator,
+    string,
+}
+
+pub trait EBNFWriter {
```

Comment:
Since almost all writers return `Result<>`, should we make these methods return a `Result<>` instead?